runtime/tests/runtime_integration_tests/test_pcr.rs

```diff
@@ -72,3 +74,136 @@ fn test_pcr_quote() {
 
     assert!(sig.verify(&resp.digest, &pkey).unwrap());
 }
+
+fn generate_mailbox_extend_pcr_req(idx: u32, pcr_extension_data: [u8; 48]) -> ExtendPcrReq {
+    let cmd = ExtendPcrReq {
+        hdr: MailboxReqHeader { chksum: 0 },
+        pcr_idx: idx,
+        data: pcr_extension_data,
+    };
+
+    let checksum = caliptra_common::checksum::calc_checksum(
+        u32::from(CommandId::EXTEND_PCR),
+        &cmd.as_bytes()[4..],
+    );
+
+    ExtendPcrReq {
+        hdr: MailboxReqHeader { chksum: checksum },
+        pcr_idx: cmd.pcr_idx,
+        data: cmd.data,
+    }
+}
+
+pub fn get_model_pcrs(model: &mut DefaultHwModel) -> [[u8; 48]; 32] {
+    let mut cmd = MailboxReq::QuotePcrs(QuotePcrsReq {
+        hdr: MailboxReqHeader { chksum: 0 },
+        nonce: [0u8; 32],
+    });
+    cmd.populate_chksum().unwrap();
+
+    let resp = model
+        .mailbox_execute(u32::from(CommandId::QUOTE_PCRS), cmd.as_bytes().unwrap())
+        .unwrap()
+        .unwrap();
+
+    return QuotePcrsResp::read_from(resp.as_slice()).unwrap().pcrs;
+}
+
+#[test]
+fn test_extend_pcr_cmd_multiple_extensions() {
+    // 0. Get fresh pcr state and verify
+    let mut model = run_rt_test(None, None, None);
+    assert_eq!(get_model_pcrs(&mut model)[4], [0u8; 48]);
+
+    // 1.0 Testing for extension_data [0,...,0]
+    let extension_data = [0u8; 48];
+
+    let cmd = generate_mailbox_extend_pcr_req(4, extension_data);
+    let res = model.mailbox_execute(u32::from(CommandId::EXTEND_PCR), cmd.as_bytes());
+    assert!(res.is_ok());
+
+    // 1.1 Checking for PCR values using PCR_QUOTE
+    let pcrs = get_model_pcrs(&mut model);
+    assert_eq!(
+        pcrs[4],
+        [
+            245, 123, 183, 237, 130, 198, 174, 74, 41, 230, 201, 135, 147, 56, 197, 146, 199, 212,
+            42, 57, 19, 85, 131, 232, 204, 190, 57, 64, 242, 52, 75, 14, 182, 235, 133, 3, 219, 15,
+            253, 106, 57, 221, 208, 12, 208, 125, 131, 23
+        ]
```

Comment:
Instead of hard-coding this (and other expected values) I think we should just compute the hashes on the test side with the `sha2` crate and compare them to the expected output.

Reply:
Since the calculation of sha2 here is depending on the hardware model, I suggest not recalculating it using the same model, because then we could potentially be comparing two erroneous results that are the same but still wrong.

The values I used in the test were calculated with a python script that uses battle-tested software implementations of the hash functions.
For further clarification, I will add the python script to this PR.

Reply:
It does not depend on the hardware model. The Quote command does something to the SHA engine resulting in a wrong result on the FPGA. There is no need to have a script just to compute basic hash values. The runtime cost of computing it is little and it's much clearer what is going on than comparing a precomputed output. Maybe rust even computes the hash at compiletime as the data is static?

Reply:
I'm just recommending you use a 3rd-party rust software crate (unrelated to Caliptra) to compute the hash of zeros instead of computing it elsewhere and pasting it here.

Fwiw, I don't think doing this will fix the CI failure you're seeing. But I do think it would make the test more readable/maintainable.

I'll try to spend some time on the FPGA today to see if I can see why this might be happening.

Reply:
Sounds good, I am working on that.

Reply:
@jhand2 CI fails due to weird behavior of the SHA engine as demonstrated here #1233

Reply:
Looks like the hardware folks were able to reproduce this bug in simulation: https://github.com/chipsalliance/caliptra-rtl/issues/375

So the proposed workaround is to re-initialize the SHA engine after doing `gen_pcr_hash` and before `pcr_extend`.

I'm thinking the easier option would be to do it at the [beginning of pcr_extend](https://github.com/chipsalliance/caliptra-sw/blob/0bbb29f66eac7d65bb606a8df766796125033300/drivers/src/sha384.rs#L180). Something like `self.sha512.regs_mut().ctrl().write(|w| w.init(true);`. I confirmed that this resolves the issue on my FPGA.

Let's do that workaround for now, and then we can decide later if we think it's worth taking the RTL fix in 1.0 or not.

Reply:
This PR should fix the problem: https://github.com/chipsalliance/caliptra-sw/pull/1249